holoviews/core/spaces.py

```diff
@@ -625,7 +648,14 @@ def _execute_callback(self, *args):
         flattened = [(k,v) for kws in kwarg_items for (k,v) in kws
                      if k not in kdims]
         with dynamicmap_memoization(self.callback, self.streams):
-            retval = self.callback(*args, **dict(flattened))
+
+            if self._posarg_keys:
+                kwargs = dict(flattened, **dict(zip(self._posarg_keys, args)))
+                args = ()
+            else:
+                kwargs = dict(flattened)
```

Comment:
Agreed. I just did this because I wanted to be close to the actual call when I wrote it.